spacepy/pybats/cimi.py

```diff
@@ -0,0 +1,343 @@
+        lines=f.readlines()
+        f.close()
+        
+        # Determine size of file.
+        nTimes=lines.count(lines[0])
+        nLat =int(lines[2].split()[0])
+        nLon =int(lines[2].split()[1]) 
+
+        self.attrs['nLat'] =nLat
+        self.attrs['nLon'] =nLon
+        self.attrs['nTime']=nTimes
+
+        nDataLength = nLat*nLon
+        
+        # Start building time array.
+        self['time']=np.zeros(nTimes, dtype=object)
+        
+        # Get variable names; pop g and rbody
+        var=(lines[4].split())[0:-2]
+        self._rawvar=var
+        
+        # initialize arrays to hold data
+        for v in var:
+            self[v]=dmarray(np.zeros((nTimes, nDataLength)))
+            
+            # Set units if possible (for now just set none
+            self[v].attrs['units']=None
+                    
+        # Loop through rest of data to fill arrays.
+        for i in range(nTimes):
+            t=float((lines[i*(nDataLength+5)+1].split())[1])
+            self['time'][i]=starttime+dt.timedelta(seconds=t)
+            for j, l in enumerate(lines[i*(nDataLength+5)+5:(i+1)*(nDataLength+5)]):
+                parts=l.split()
+                for k,v in enumerate(var):
+                    self[v][i,j]=float(parts[k])
+                            
+
+    def add_slice(self, var, time, nlev=31, zlim=None, xlim=None, ylim=None,
+                  target=None, 
+                  loc=111, title=None,
+                  add_cbar=True, clabel=None,
+                  show_pts=False, dolog=False, add_body=True,  
+                  figsize=(8.34,7),
+                  *args, **kwargs):
+        '''
+        Create a plot of variable *var* at a given time.
+
+        If kwarg **target** is None (default), a new figure is 
+        generated from scratch.  If target is a matplotlib Figure
+        object, a new axis is created to fill that figure at subplot
+        location **loc**.  If **target** is a matplotlib Axes object, 
+        the plot is placed into that axis.
+        '''
+
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import (LogNorm, Normalize)
+        from matplotlib.patches import Circle
+        from matplotlib.ticker import (LogLocator, LogFormatter, 
+                                       LogFormatterMathtext, MultipleLocator)
+        from matplotlib.patches import Wedge
+        
+        # Grab the slice of data that we want:
+        if type(time) == type(self['time'][0]):
+            if time not in self['time']:
+                raise ValueError('Time not in object')
+            time = np.arange(self.attrs['nTime'])[self['time']==time][0]
+
+        fig, ax = set_target(target, loc=loc, figsize=figsize)
+        ax.set_aspect('equal')
+
+        # Grab values from correct time/location.
+        x = self['x'][time, :]
+        y = self['y'][time, :]
+        value = self[var][time, :]
+
+        # Get max/min if none given.
+        if zlim==None:
```

Comment:
This comparision should be [is](https://peps.python.org/pep-0008/#programming-recommendations).